Prose/ProseLib/Sources/App/AppScene.swift

```diff
@@ -35,6 +35,7 @@ public struct AppScene: Scene {
     }
 
     public var body: some Scene {
+        let _ = actions.send(.onAppear)
```

Comment:
That should certainly go into the `.onAppear(perform:)` handler.

Reply:
No, because I want it to be called only once, and not once per window.

I didn't like it either and with #38 it [is now fixed](https://github.com/prose-im/prose-app-macos/pull/38/commits/1592ec7cd70284823401eed611b64c471987b488#diff-7e699972e390677c42757230305df7bd9805efc806d35b0e57f362e3e99cd06aR46).

Reply:
Now you're calling it everytime `body` gets invalidated which you have fairly little control over. If you want to be certain, add a `isVisible: Bool` property to the AppState and ignore `.onAppear` actions there if is already true.

Reply:
Ok, I will do that but in the other PR (referencing the comment there)